src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10385,6 +10385,7 @@ private static bool CanFollowCast(SyntaxKind kind)
                 case SyntaxKind.MinusMinusToken:
                 case SyntaxKind.OpenBracketToken:
                 case SyntaxKind.DotToken:
+                case SyntaxKind.DotDotToken:
                 case SyntaxKind.MinusGreaterThanToken:
                 case SyntaxKind.QuestionQuestionToken:
                 case SyntaxKind.EndOfFileToken:
```

Comment:
note: i don't like this code here.  the spec is written clearly as a whitelist of tokens that cause us to think something is a cast expression.  However, the parser is written as a blacklist.  Meaning that any time we add new tokens (like dotdot) we need to know to fix things up here.  I'd really like to just make this more in line with the rules in https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#cast-expressions, but i'm worried about the risk of that.  If people want, i'd be willing to make another PR that tries that.

Reply:
@gafter for thoughts here.